add align to barge method

R1shy opened the pull request:
BC Link: https://3.basecamp.com/5677500/buckets/34348388/card_tables/cards/8904511350

Commit:
.

Commit:
.

R1shy committed:
add align to barge method

R1shy committed:
.

R1shy committed:
fixes

R1shy committed:
fixes

No767 commented on .vscode/settings.json:
```diff
@@ -57,6 +57,6 @@
     "edu.wpi.first.math.**.struct.*",
   ],
-  "java.jdt.ls.vmargs": "-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx4G -Xms100m -Xlog:disable",
+  "java.jdt.ls.vmargs": "-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx8G -Xms100m -Xlog:disable",
```
Let's keep this at 4G instead, as we know that it works. but im wondering why it's necessary to change it?

R1shy committed:
fix ram allocation

RobertTheBlair commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -499,6 +495,9 @@ private void configureBindings() {
     //     .and(m_driverController.a())
     //     .whileTrue(alignToPose(() ->
     // EagleUtil.getClosestLeftReefBack(drivetrain.getPose(0.25))));
+    IS_L4
```
Why does this need `IS_L4` ?

No767 replied:
most likely it should be on `IS_L1` instead

R1shy replied:
Done

RobertTheBlair replied:
I was curious why the target of the coral matters at all when aligning to the barge.

R1shy commented:
Just for testing if you think this s ready to merge i can remove it

On Mon, Aug 25, 2025, 8:51 PM Rob Blair ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/frc/robot/RobotContainer.java
> <https://github.com/gwhs/2025-Reefscape/pull/311#discussion_r2299678837>:
>
> > @@ -499,6 +495,9 @@ private void configureBindings() {
>      //     .and(m_driverController.a())
>      //     .whileTrue(alignToPose(() ->
>      // EagleUtil.getClosestLeftReefBack(drivetrain.getPose(0.25))));
> +    IS_L4
>
> Why does this need IS_L4 ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gwhs/2025-Reefscape/pull/311#pullrequestreview-3153781694>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIUCBKUIP2VXOPAPOO7VLET3PPKVFAVCNFSM6AAAAACEZJC27WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTCNJTG44DCNRZGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

R1shy commented:
Ok, there is a different command linked to that should I add or replace

On Mon, Aug 25, 2025, 9:14 PM Noelle Wang ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/frc/robot/RobotContainer.java
> <https://github.com/gwhs/2025-Reefscape/pull/311#discussion_r2299702632>:
>
> > @@ -499,6 +495,9 @@ private void configureBindings() {
>      //     .and(m_driverController.a())
>      //     .whileTrue(alignToPose(() ->
>      // EagleUtil.getClosestLeftReefBack(drivetrain.getPose(0.25))));
> +    IS_L4
>
> most likely it should be on IS_L1 instead
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gwhs/2025-Reefscape/pull/311#discussion_r2299702632>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIUCBKTK4BCGCLJXEGDTHPT3PPNJVAVCNFSM6AAAAACEZJC27WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTCNJTHAYTCNJTHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

R1shy committed:
fix trigger for alignment

R1shy committed:
base align off position like issac wanted

R1shy committed:
.

RobertTheBlair commented on src/main/java/frc/robot/EagleUtil.java:
```diff
@@ -566,4 +567,13 @@ public static double getRotationCenterReef(Pose2d pose) {
+
+  public static Pose2d aligntobarge(CommandSwerveDrivetrain drive) {
+    if (FieldConstants.HALF_THE_FIELD > drive.getPose().getX()) {
+      Pose2d poseBlue = new Pose2d(8, drive.getPose().getY(), Rotation2d.fromDegrees(0));
```
Should 8 and 9.6 be in FieldConstants?

RobertTheBlair commented on src/main/java/frc/robot/EagleUtil.java:
```diff
@@ -566,4 +567,13 @@ public static double getRotationCenterReef(Pose2d pose) {
     return reef.getTranslation().minus(pose.getTranslation()).getAngle().getDegrees();
   }
+
+  public static Pose2d aligntobarge(CommandSwerveDrivetrain drive) {
```
If you pass in the current pose2d, then you can test it standalone from the drivetrain.  Makes the function more flexible

RobertTheBlair commented on src/main/java/frc/robot/RobotContainer.java:
we still want some way to invoke the align to barge.   I think hooking it to right bumper may be all we need in the end, not worrying about the is level 1 or 4

No767 replied:
based on the new controller bindings, it would most likely be left bumper. hold left bumper would auto align to the net

R1shy committed:
.

isaacngkh commented on src/main/java/frc/robot/FieldConstants.java:
```diff
@@ -4,11 +4,11 @@
   public static List<Pose2d> redAlgaeSetpointList = EagleUtil.calculateRedAlgaeSetPoints();
+  public static double HALF_THE_FIELD = 8.75;
+  public static int ON_LEFT = 8;
+  public static double ON_RIGHT = 9.6;
```
on left and on right are kinda not quiet descriptive enough. How bout something like SCORE_NET_BLUE_SIDE_X and SCORE_NET_RED_SIDE_X? also double instead of int.

isaacngkh commented on src/main/java/frc/robot/EagleUtil.java:
```diff
@@ -566,4 +567,13 @@ public static double getRotationCenterReef(Pose2d pose) {
+      Pose2d poseBlue = new Pose2d(FieldConstants.ON_LEFT, pose.getY(), Rotation2d.fromDegrees(0));
+      return poseBlue;
+    }
+    Pose2d poseRed = new Pose2d(FieldConstants.ON_RIGHT, pose.getY(), Rotation2d.fromDegrees(180));
```
I looked at the net, I think when scoring in the net on the blue side, it should be 0 deg and when on the red side, it should be 180 deg. So just need to flip

R1shy committed:
Merge branch 'dev' into score-in-net-aligb

R1shy committed:
.

R1shy committed:
update @ Thu Sep  4 07:29:43 PM PDT 2025

RobertTheBlair reviewed (dismissed):
Be sure to test this on a real robot

Commit:
-`
                 .o+`
                `ooo/
               `+oooo:
              `+oooooo:
              -+oooooo+:
            `/:-:++oooo+:
           `/++++/+++++++:
          `/++++++++++++++:
         `/+++ooooooooooooo/`
        ./ooosssso++osssssso+`
       .oossssso-````/ossssss+`
      -osssssso.      :ssssssso.
     :osssssss/        osssso+++.
    /ossssssss/        +ssssooo/-
  `/ossssso+/:-        -:/+osssso+-
 `+sso+:-`                 `.-/+oso:
`++:.                           `-/+/
.`                                 `/[1G[18A[41Crishy@EMMABOOK
[41C--------------
[41COS: Arch Linux ARM aarch64
[41CHost: Apple MacBook Air (13-inch, M2, 2022)
[41CKernel: Linux 6.16.4-asahi-2-1-ARCH
[41CUptime: 1 hour, 21 mins
[41CPackages: 873 (pacman)
[41CShell: bash 5.3.3
[41CWM: Hyprland 0.49.0 (Wayland)
[41CCursor: Adwaita
[41CTerminal: kitty 0.42.2
[41CTerminal Font: HackNFM-Regular (11pt)
[41CCPU: Apple M2 (8) @ 3.20 GHz
[41CGPU: Apple M2 (10) @ 1.40 GHz [Integrated]
[41CMemory: 6.40 GiB / 23.03 GiB (28%)
[41CSwap: Disabled
[41CDisk (/): 29.78 GiB / 832.17 GiB (4%) - ext4
[41CLocal IP (wlan0): 10.186.67.200/20
[41CLocale: en_US.UTF-8
[41C
